### PR TITLE
Added context lost explainer updates

### PR DESCRIPTION
**Pull request by macrabil**
Here is a set of changes to match the current changes/behavior for Canvas2dContextLostRestored in Chromium.

### CODE REVIEW

**Review by travisleithead — APPROVED** [review]
This seems to match what we reviewed in the code. Thanks Matthew.

File: spec/context-loss.md [hunk 1]
```diff
@@ -34,14 +34,18 @@ When the user agent detects that the backing storage associated with a Canvas2D
     1. Fire a event `contextlost` at *canvas*
     2. If the event's canceled flag is not set, abort these steps.
```
Comment (travisleithead) [inline spec/context-loss.md:35]: Oh, I think this logic is backwards (from what's in the code, and what has been discussed previously) and needs to be changed. (2D context behavior is opposite the behavior of WebGL context lost for this case.): So `If the event's canceled flag is set, abort these steps` and then add some kind of a description around not restoring the canvas (and what that means).

**Review by travisleithead — CHANGES_REQUESTED** [review]
Missed the cancelation behavior.

**Review by travisleithead — APPROVED** [review]
Super. Thanks for the update!

File: spec/context-loss.md [hunk 2]
```diff
@@ -32,16 +32,20 @@ When the user agent detects that the backing storage associated with a Canvas2D
 3. Set *context lost* flag.
 4. Queue a task to perform the following steps:
     1. Fire a event `contextlost` at *canvas*
-    2. If the event's canceled flag is not set, abort these steps.
+    2. If the event's canceled flag is set, abort these steps. The backing buffer for the context will not be restored.
```
Comment (fserb) [inline spec/context-loss.md:35]: what is the use case for not restoring the buffer?

File: spec/context-loss.md [hunk 3]
```diff
@@ -32,16 +32,20 @@ When the user agent detects that the backing storage associated with a Canvas2D
...
 
 UA could add support for "context loss test" by creating a
 `console.resetGraphicsContext()` method.
 
 ### Open issues and questions
 
 - Deprecate `webglcontextlost` in favor of `contextlost`?
+- Deprecate `webglcontextrestored` in favor of `contextrestored`?
 - What the default behavior should be?
+- Currently the canvas has to be modified in order for context lost/restored events to fire. Is that the right behavior?
```
Comment (fserb) [inline spec/context-loss.md:47]: I think that makes sense, as we don't build the buffer on canvas creation

File: spec/context-loss.md [hunk 4]
```diff
@@ -32,16 +32,20 @@ When the user agent detects that the backing storage associated with a Canvas2D
...
+- The canvas has to be in the DOM and also visible in order for context lost/restored events to fire. Is that the right behavior?
```
Comment (fserb) [inline spec/context-loss.md:48]: hmmmmm. I think it should trigger even if the canvas is not visible. Also, it should trigger for `OffscreenCanvas` too.

File: spec/context-loss.md (hunk 2 above)
Comment (travisleithead) [reply spec/context-loss.md:35]: My understanding is that some applications are using extra canvases as distributed buffers for assembling large scenes. The use case here is that if the context fails (potentially due to GPU OOM) then the app will use the cancelation behavior to prevent the UA from "restarting" the "extra" canvases that may have contributed to the memory pressure in the first place (the app will dynamically scale down the number of extra distributed canvases in use in this scenario, possibly to zero).

In short, the flag optimizes the case where the canvas will no longer be used again, so the UA won't need to bother allocating and resetting it just so the app will then remove it and have it be GC'd later.

@macrabil does that sound right to you?

File: spec/context-loss.md (hunk 4 above)
Comment (travisleithead) [reply spec/context-loss.md:48]: Agree. I think the canvas' connectedness shouldn't matter as far as the firing of the event is concerned.

File: spec/context-loss.md (hunk 2 above)
Comment (macrabil) [reply spec/context-loss.md:35]: Sounds right, yeah.

Comment (fserb) [reply spec/context-loss.md:35]: Makes sense.
